### PR TITLE
Add msal

### PR DESCRIPTION
**Pull request by marosmars**
Changes:

- Make /dashboard the root URL for dashboard APP (both in app and in server)
- Add msal login/logout/refresh to enable integration with Microsoft AD identity provider
- Make msal integration optional
- Make dashboard configurable via .env ... including the links from dashboard
- Add simple notification mechanism to dashboard in order to "redirect to dashboard with a message" e.g. when api-gateway detects unauthorized user, redirect them back to dashboard with appropriate message

Note: **Everything here should be backwards compatible EXCEPT the base url change to /dashboard**

### CODE REVIEW

**Comment by marosmars** [comment]
image
image

File: app/.env [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+REACT_APP_AUTH_ENABLED=false
+REACT_APP_AD_CLIENT_ID=yyyyyyyyyyyyyyyyyyyyyyyyy
+REACT_APP_AD_REDIRECT_URL=http://localhost:3000/
+
+# Dashboard items config
+REACT_APP_URL_INVENTORY='window.location.protocol + "//" + window.location.hostname + ":5601"'
+REACT_APP_URL_INVENTORY_ENABLED=true
+
+REACT_APP_URL_UNICONFIG=/uniconfig/ui
+REACT_APP_URL_UNICONFIG_ENABLED=true
```
Comment (Paulooze) [inline app/.env:15]: What happens when `REACT_APP_URL_UNICONFIG_ENABLED=true` and `REACT_APP_URL_UNICONFIG` is empty?

File: app/src/dashboard/panel/Panel.js [hunk 2]
```diff
@@ -7,7 +7,13 @@ function Panel(props) {
   const [highlight, setHighlight] = useState(false);
 
   return (
-    <div onClick={() => window.location = props.link}>
+    <div onClick={() => {
```
Comment (Paulooze) [inline app/src/dashboard/panel/Panel.js:10]: Why do it this way? Can't we have a simple `<a href={props.link}" target={isExtermal ? '_blank' : undefined}>`?

**Comment by marosmars** [comment]
@Paulooze thx for review / improvements. @ChmaraX pls take a look as well, as the dashboard is moving from / to /dashboard root URL which affects FM / demo.frinx.io ?

**Review by Paulooze — APPROVED** [review]

**Review by ChmaraX — APPROVED** [review]
It looks good to me from code standpoint. 

@marosmars demo.frinx.io and FM probably doesn't matter since FM will be replaced by upcoming docker-swarm deployment, where each component has its own release, so dashboard will be tested/adjusted when new deployment will be ready to be released. So its possible that this change wont be present in first release of "new" FM to make release quicker. 
@MartinSunal am I right?

**Pull request merged**